src/map/utils/battleutils.cpp

```diff
@@ -730,6 +730,16 @@ namespace battleutils
 
                         if (PDefender->isAlive())
                         {
+                            // Subpower is the remaining damage that can be drained. When it reaches 0 the effect ends
+                            int remainingDrain = PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_DREAD_SPIKES)->GetSubPower();

```

Comment:
GetStatusEffect can return nullptr, check it

there is no guarantee that the effect is the only thing that set action.spikesParam to SPIKE_DREAD or SPIKE_REPRISAL

auto PEffect = PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_DREAD_SPIKES)
if (PEffect) {